.teamcity/settings.kts

```diff
@@ -31,14 +31,25 @@ To debug in IntelliJ Idea, open the 'Maven Projects' tool window (View
 version = "2019.2"
 
 project {
+    buildType {
+        name = "Verify"
+        id = RelativeId(name.toId())
+        vcs {
+            root(DslContext.settingsRootId)
+            cleanCheckout = true
```

Comment:
For the release build the rationale was that the maven-release-plugin creates quite a few temporary files and we wanted to ensure those don't influence the build. The same holds true for the CI build because it might run on the same node as a previously failed release build.

Reply:
We could have a step that calls `release:clean` (https://maven.apache.org/maven-release/maven-release-plugin/clean-mojo.html) in the deploy to maven central pipeline. That way we could remove the clean checkout in the verify build. I would prefer to keep it in the release build.

@guylabs would you prefer that?

Reply:
Ok to keep it, but I'd add a small comment stating why we use a clean checkout, as this normally adds unnecessary work to the build which increases build time.

A `release:clean` would be more precise as we would only clean the release plugin files. But I leave that up to you.

Reply:
@britter That step would have to run even if the released failed, though, right?

Reply:
I've added a comment for now since I don't expect this to run often and if it runs it will be fairly quick even with the clean checkout. We can revisit if this becomes a bottleneck.